### PR TITLE
Use error thresholds by default

### PR DESCRIPTION
**Pull request by rheacangeo**
## Purpose
The validation error thresholds defined by the Translate classes can be overridden for specific backends and platforms using a yaml file, a collection of relaxed thresholds we should eventually investigate the reasons for and either or officially relax them in the class or fix a bug. This was initially implemented just for the jenkins tests, to enable automated testing to work, with the thought local developers would choose whether or not to use an override file. But, for consistency and convenience, this PR moves the specification of the thresholds file into the Makefile. This makes experimenting with other datasets more strict. Now for the Makefile to work properly, one would need to name the experiment with the same `c##_#ranks_<>` convention existing experiments use, and have an existing overrides filename that matches the `<>` in the experiment name. This is less flexible, but more convenient for our main use case. 

## Infrastructure changes:

- Removed specification of overrides thresholds in .jenkins/jenkins.sh
- Removed the profile variables in the .jenkins/jenkins.sh script leftover from a previous 
- Adds the --threshold_overrides_file argument to the pytest commands in the Makefile, that will pick 'standard' or 'baroclinic' based on what EXPERIMENT is set to 

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by rheacangeo** [comment]
launch jenkins

File: Makefile [hunk 1]
```diff
@@ -30,10 +30,13 @@ CORE_TAR=$(SARUS_FV3CORE_IMAGE).tar
 MPIRUN_CALL ?=mpirun -np $(NUM_RANKS)
 BASE_INSTALL?=$(FV3)-install-serialbox
 DATA_BUCKET= $(REGRESSION_DATA_STORAGE_BUCKET)/$(FORTRAN_SERIALIZED_DATA_VERSION)/$(EXPERIMENT)/
-DEV_MOUNTS = '-v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $(FV3UTIL_DIR):/usr/src/fv3gfs-util -v $(TEST_DATA_HOST):$(TEST_DATA_RUN_LOC)'
-PYTEST_SEQUENTIAL=pytest --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS) $(FV3_PATH)/tests
-PYTEST_PARALLEL=$(MPIRUN_CALL) pytest --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS) -m parallel $(FV3_PATH)/tests
 
+DEV_MOUNTS = '-v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $(FV3UTIL_DIR):/usr/src/fv3gfs-util -v $(TEST_DATA_HOST):$(TEST_DATA_RUN_LOC) '
+TEST_TYPE=$(word 3, $(subst _, ,$(EXPERIMENT)))
```
Comment (eddie-c-davis) [inline Makefile:35]: Bash ninja!

File: Makefile [hunk 2]
```diff
@@ -30,10 +30,13 @@ CORE_TAR=$(SARUS_FV3CORE_IMAGE).tar
 MPIRUN_CALL ?=mpirun -np $(NUM_RANKS)
 BASE_INSTALL?=$(FV3)-install-serialbox
 DATA_BUCKET= $(REGRESSION_DATA_STORAGE_BUCKET)/$(FORTRAN_SERIALIZED_DATA_VERSION)/$(EXPERIMENT)/
-DEV_MOUNTS = '-v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $(FV3UTIL_DIR):/usr/src/fv3gfs-util -v $(TEST_DATA_HOST):$(TEST_DATA_RUN_LOC)'
-PYTEST_SEQUENTIAL=pytest --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS) $(FV3_PATH)/tests
-PYTEST_PARALLEL=$(MPIRUN_CALL) pytest --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS) -m parallel $(FV3_PATH)/tests
 
+DEV_MOUNTS = '-v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $(FV3UTIL_DIR):/usr/src/fv3gfs-util -v $(TEST_DATA_HOST):$(TEST_DATA_RUN_LOC) '
```
Comment (eddie-c-davis) [inline Makefile:34]: Probably has no impact, but is the added space at the end here intentional?

File: README.md [hunk 3]
```diff
@@ -146,6 +146,8 @@ common options for our tests, which you can add to `TEST_ARGS`:
 
 * `-m` - will let you run only certain groups of tests. For example, `-m=parallel` will run only parallel stencils, while `-m=sequential` will run only stencils that operate on one rank at a time.
 
+* `--threshold_overrides_file` - will read a yaml file with error thresholds specified for specific backend and platform (docker or metal) configurations, overriding the max_error thresholds defined in the Translate classes.
```
Comment (eddie-c-davis) [inline README.md:149]: Way to update the docs!

**Review by eddie-c-davis — APPROVED** [review]
One small question, but looks great!

File: Makefile (hunk 2 above)
Comment (rheacangeo) [reply Makefile:34]: oh oops -- wasn't intentional :)

**Pull request merged**